### PR TITLE
Encode the authentication reject error to build a proper callback url

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Encode the `&error_code=access_denied` query param for the callback url in order to fix the bug when the authentication request appears again if it was rejected.

**DO NOT MERGE** until https://github.com/eclipse-che/che-dashboard/pull/932 is merged
### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22499
fixes https://github.com/eclipse/che/issues/22539

### How to test this PR?

1. Deploy che with the PR image `quay.io/eclipse/che-server:pr-568`
2. Setup the [GitHub oauth flow](https://eclipse.dev/che/docs/stable/administration-guide/configuring-oauth-2-for-github/).
3. Create a workspace from a public GitHub repository.
4. When the authentication page appears, reject the request.

See: the workspace start proceeds without fetching the token. 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
Please find the error below:
screencast-bpconcjcammlapcogcnnelfmaeghhagj-2023 09 21-13_06_52

**Comment by vinokurig** [comment]
@tolusha 
> Please find the error below

This is another bug. You can reproduce it even on dogfooding:
1. Start a workspace from a GitHub repository, accept the authorisation as you can not reject it because of this bug.
2. Stop the workspace.
3. Go to dashboard -> user preferences -> git services and revoke the authorisation.
4. Start the workspace and reject the authorisation.

I don't think we should do something with this because according to the issue we will not ask to authorise on workspace start, as the authorisation can be either accepted or rejected once on workspace create step.

**Comment by vinokurig** [comment]
@tolusha opened a separate issue: https://github.com/eclipse/che/issues/22539

**Review by tolusha — APPROVED** [review]

File: wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java [hunk 1]
```diff
@@ -84,9 +87,18 @@ public Response callback(UriInfo uriInfo, List<String> errorValues) throws NotFo
...
+        String query = url.getQuery();
+        return Response.temporaryRedirect(
+                URI.create(
+                    redirectAfterLogin.substring(0, redirectAfterLogin.indexOf(query))
+                        + URLEncoder.encode(query + "&error_code=access_denied", UTF_8)))
+            .build();
+      } catch (MalformedURLException e) {
+        LOG.error(e.getMessage(), e);
+        throw new RuntimeException(e);
+      }
```
Comment (ibuziuk) [inline wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:101]: can we extract it to the ac separate method and add a doc for it?

**Review by ibuziuk — APPROVED** [review]

**Comment by artaleks9** [comment]
Verified on Che deployed with
```
quay.io/eclipse/che-server:pr-568
quay.io/eclipse/che-dashboard:pr-932
```
The functionality works as expected.

**Review by artaleks9 — APPROVED** [review]

**Pull request merged**